crates/aptos/src/account/create_resource_account.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::types::{CliCommand, CliTypedResult, TransactionOptions};
+use aptos_rest_client::{
+    aptos_api_types::{WriteResource, WriteSetChange},
+    Transaction,
+};
+use aptos_transaction_builder::aptos_stdlib;
+use aptos_types::account_address::AccountAddress;
+use async_trait::async_trait;
+use clap::Parser;
+use serde::Serialize;
+
+/// Command to create a resource account
+///
+#[derive(Debug, Parser)]
+pub struct CreateResourceAccount {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+
+    /// Resource account seed.
+    #[clap(long)]
+    pub(crate) seed: String,
+
+    /// Optional Resource Account authentication key.
+    #[clap(long, parse(try_from_str = crate::common::types::load_account_arg))]
+    pub(crate) authentication_key: Option<AccountAddress>,
+}
+
+/// A shortened create resource account output
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct CreateResourceAccountSummary {
+    pub gas_used: Option<u64>,
+    pub sender: Option<AccountAddress>,
+    pub resource_account: Option<AccountAddress>,
+    pub hash: Option<String>,
+    pub success: bool,
+    pub version: Option<u64>,
+    pub vm_status: String,
+}
+
+impl From<Transaction> for CreateResourceAccountSummary {
```

Comment:
This is a tricky one, the API, should always return the correct transaction, so we should probably use try_from, and throw a CliError if it doesn't have the necessary information.

Reply:
I was looking at the implementation of `transfer` and it uses `from`.